src/robust_provider.rs

```diff
@@ -78,6 +87,21 @@ impl<N: Network> RobustProvider<N> {
         self
     }
 
+    /// Get a reference to the inner provider
+    #[must_use]
+    pub fn inner(&self) -> &RootProvider<N> {
+        &self.provider
+    }
+
+    /// Add a fallback provider to the list.
+    ///
+    /// Fallback providers are used when the primary provider times out.
+    #[must_use]
+    pub fn fallback_provider(mut self, provider: RootProvider<N>) -> Self {
```

Comment:
```suggestion
    pub fn fallback(mut self, provider: RootProvider<N>) -> Self {
```
nit: clear enough + succinct